micro-rdk/src/common/conn/server.rs

```diff
@@ -385,6 +389,7 @@ where
             app_client_tasks,
             network,
             rpc_host,
+            _mdns: mdns,
```

Comment:
If we lose the network, we may need to re-create this since we may come up with a different IP. Maybe that happens organically as part of refactoring the relationship between networking/robot/provisioning/server? I'm not inclined to tackle this directly in this review because I suspect there are other places where we have this problem as well. Should I file a ticket so we don't lose track of the need to handle that better?

Reply:
I'd like to track it. We can maybe put network, mdns, and whatever else might be coupled to it in a single struct and simplify these arguments.